tests/foreman/api/test_activationkey.py

```diff
@@ -50,470 +50,482 @@ def _bad_max_hosts():
+
+    # Disassociate host collection from the activation key.
+    act_key.remove_host_collection(data={'host_collection_ids': [host_collection.id]})
+    assert len(act_key.read().host_collection) == 0
+
+
+@pytest.mark.tier1
+def test_positive_update_auto_attach():
+    """Create an activation key, then update the auto_attach
+    field with the inverse boolean value.
+
+    :id: ec225dad-2d27-4b37-989d-1ba2c7f74ac4
+
+    :expectedresults: The value is changed.
+
+    :CaseImportance: Critical
+    """
+    act_key = entities.ActivationKey().create()
+    act_key_2 = entities.ActivationKey(
+        id=act_key.id, auto_attach=(not act_key.auto_attach)
+    ).update(['auto_attach'])
+    assert act_key.auto_attach != act_key_2.auto_attach
+
+
+@pytest.mark.tier1
+@pytest.mark.upgrade
+@pytest.mark.parametrize('name', **parametrized(valid_data_list()))
+def test_positive_delete(name):
+    """Create activation key and then delete it.
+
+    :id: aa28d8fb-e07d-45fa-b43a-fc90c706d633
+
+
+    :expectedresults: Activation key is successfully deleted.
+
+    :CaseImportance: Critical
+
+    :parametrized: yes
+    """
+    act_key = entities.ActivationKey().create()
+    act_key.delete()
+    with pytest.raises(HTTPError):
+        entities.ActivationKey(id=act_key.id).read()
+
+
+@pytest.mark.tier2
+def test_positive_remove_user():
+    """Delete any user who has previously created an activation key
+    and check that activation key still exists
+
+    :id: 02ce92d4-8f49-48a0-bf9e-5d401f84cf46
+
+    :expectedresults: Activation Key can be read
+
+    :BZ: 1291271
+    """
+    password = gen_string('alpha')
+    user = entities.User(password=password, login=gen_string('alpha'), admin=True).create()
+    cfg = get_nailgun_config()
+    cfg.auth = (user.login, password)
+    ak = entities.ActivationKey(cfg).create()
+    user.delete()
+    try:
+        entities.ActivationKey(id=ak.id).read()
+    except HTTPError:
+        pytest.fail("Activation Key can't be read")
+
+
+@pytest.mark.upgrade
+@pytest.mark.run_in_one_thread
+@skip_if_not_set('fake_manifest')
+@pytest.mark.tier2
+def test_positive_fetch_product_content(module_org):
+    """Associate RH & custom product with AK and fetch AK's product content
+
+    :id: 424f3dfb-0112-464b-b633-e8c9bce6e0f1
+
+    :expectedresults: Both Red Hat and custom product subscriptions are
+        assigned as Activation Key's product content
+
+    :BZ: 1426386
+
+    :CaseLevel: Integration
+
+    :CaseImportance: Critical
+    """
+    with manifests.clone() as manifest:
+        upload_manifest(module_org.id, manifest.content)
+    rh_repo_id = enable_rhrepo_and_fetchid(
+        basearch='x86_64',
+        org_id=module_org.id,
+        product=PRDS['rhel'],
+        repo=REPOS['rhst7']['name'],
+        reposet=REPOSET['rhst7'],
+        releasever=None,
+    )
+    rh_repo = entities.Repository(id=rh_repo_id).read()
+    rh_repo.sync()
+    custom_repo = entities.Repository(
```

Comment:
Setting up custom repo can be used by a fixture, but this isn't necessary at this time.  Just not and be aware that it could reduce code here.